### ISSUE TITLE
Seznam bot

### ISSUE THREAD

**Issue by mikeruss1** [CIDRAM/CIDRAM#420]
this is being blocked, but is not tracked, presumably because its a search engine, its verified as from Seznam

is this what you intend ?

User agent: Mozilla/5.0 (compatible; SeznamBot/4.0-RC1; +http://napoveda.seznam.cz/seznambot-intro/) 
Signatures reference: bobuam.php:L98, bobuam.php:L118, bobuam.php:L132
Why blocked: Suspected bot or scraper probe, Suspected bot or scraper probe, Suspected bot or scraper probe! [»]
Infractions: 0
Verified identity: SeznamBot
Hostname: fulltextrobot-77-75-77-119.seznam.cz

**Comment by eurobank** [comment]
That module (bobuam) gives a LOT of wrong blocks. That is why i don't use it.

**Comment by Maikuolan** [comment]
Fixed. :-)

> presumably because its a search engine, its verified as from Seznam

Yep.

> is this what you intend ?

Nope, definitely don't intend to block Seznam, nor any other verified search engines for that matter.

Unfortunately, because the signatures in BOBUAM for blocking suspected bots and scraper probes are rather heavy (i.e., many of them, and blocking a *lot* of different user agents), and were not written by me personally (remembering that, although I'm continuing the support and maintenance of BOBUAM for CIDRAM v3, I'm not the original author/creator of BOBUAM; MacMathan is the original author/creator of BOBUAM, who has since discontinued support/maintenance; and, even then, although MacMathan wrote many of the signatures in BOBUAM, many more of them also came from a now-dated subset of the data from StopBadBots.com, authored/created by W. Minozzi).

That effectively means that some of the data included in BOBUAM could be considered as, not even secondhand, but rather thirdhand data, and there are two important implications in that: Firstly, the good: In theory, it's possible that BOBUAM may block a number of unwanted bots and nefarious entities that I've never even heard of before myself, given that there's been more eyes on its data, giving it a potentially wider scope, and potentially covering any blind spots I might've had in regards to some of the available modules which I've written myself, such as the optional security extras module and bot user agents module (i.e., more "true positives" or "correct inferences"). Secondly, the bad: Being thirdhand data, I'm not going to be intimately familiar with absolutely everything BOBUAM is capable of blocking, and as such, by using it, it may pose as an additional blind spot in its own right (i.e., a greater risk of more "false positives"). Signatures tend to imply what their author believes should be blocked, and if different authors have different opinions regarding what should/shouldn't be blocked, or if different authors have access to different information, then naturally, they'll end up making different decisions regarding the signatures they write, and one mightn't always agree with what the other writes, so what might be considered as a "correct inference" for one, might be considered as a "false positive" for the other. But, not always being intimately familiar with absolutely all of those signatures means that such false positives mightn't always be caught prior to being actually encountered.

Basically, I can't promise that the module won't cause more such false positives in the future. But, when they're caused, as long as they're reported here, I should be able to easily fix them the same as was done in this case.

**Comment by mikeruss1** [comment]
I reckon we need the module, the package needs to block bots. There are issues, I think the default outdated browser versions are too aggressive. And we can get problems fixed, worth putting up with the odd problem I think.

**Comment by mikeruss1** [comment]
Maikuolan - we did discuss a while ago sharing an API or incorporating his material in CIDRAM - nothing come of that?

**Comment by Maikuolan** [comment]
> Maikuolan - we did discuss a while ago sharing an API or incorporating his material in CIDRAM - nothing come of that?

Are you referring to the South Korea and China blocklists, or something else? You may need to remind me, sorry.

**Comment by mikeruss1** [comment]
sorry
StopBadBots.com, authored/created by W. Minozzi).

**Comment by eurobank** [comment]
**Not fixed:**

ID: 1677267027-247321-4462613311 »
Date/Time: Fri, 24 Feb 2023 21:30:27 +0200 »
IP address: 77.75.79.31 »
User agent: Mozilla/5.0 (compatible; SeznamBot/4.0-RC1; +http://napoveda.seznam.cz/seznambot-intro/) »
Signatures count: 2 »
Signatures reference: bobuam.php:L118, bobuam.php:L132 »
Why blocked: Suspected bot or scraper probe, Suspected bot or scraper probe! »

Infractions: 0
Country code lookup: XX »
Verified identity: SeznamBot »
Request method: GET »
Hostname: fulltextrobot-fresh-77-75-79-31.seznam.cz »

**Comment by Maikuolan** [comment]
Wasn't expecting three different signatures to all be falsing on the same user agent in three different ways. The previous fix fixed just one of the signatures, but not the two other. Should all be fixed properly now though (fingers crossed).

**Comment by Maikuolan** [comment]
> sorry
> StopBadBots.com, authored/created by W. Minozzi).

Hadn't really looked into it much yet (just busy with other stuff), so not sure right now. Doubtful, but I'll investigate a bit further when I get some other stuff finished first.
